colmena/models.py

```diff
@@ -146,6 +146,39 @@ def args(self) -> Tuple[Any]:
     def kwargs(self) -> Dict[str, Any]:
         return self.inputs[1]
 
+    def json(self, **kwargs: Dict[str, Any]) -> str:
+        """Override json encoder to use a custom encoder with proxy support"""
+        if 'exclude' in kwargs:
+            # Make a shallow copy of the user passed excludes
+            user_exclude = kwargs['exclude'].copy()
+            if isinstance(kwargs['exclude'], dict):
+                kwargs['exclude'].update({'inputs': True, 'value': True})
+            if isinstance(kwargs['exclude'], set):
+                kwargs['exclude'].update({'inputs', 'value'})
+            else:
+                raise ValueError(f'Unsupported type {type(kwargs["exclude"])} for argument "exclude". Expected set or dict')
+        else:
+            user_exclude = set()
+            kwargs['exclude'] = {'inputs', 'value'}
+
+        # Use pydantic's encoding for everything except `inputs` and `values`
+        data = super().dict(**kwargs)
+
+        # Add inputs/values back to data unless the user excluded them
+        if isinstance(user_exclude, set):
+            if 'inputs' not in user_exclude:
+                data['inputs'] = self.inputs
+            if 'value' not in user_exclude:
+                data['value'] = self.value
+        elif isinstance(user_exclude, dict):
+            if not user_exclude['inputs']:
+                data['inputs'] = self.inputs
+            if not user_exclude['value']:
+                data['value'] = self.value
+
+        # Jsonify with custom proxy encoder
+        return json.dumps(data, default=proxy_json_encoder)
```

Comment:
pydantic's json utilities won't work on any proxies (it always resolves them even with my custom encoder) so we separate just the `inputs` and `values` attributes. All other attributes get jsonified with pydantic on line 165.

I also added some checks to properly handle the user passing `exclude=...`.